cuebot/scheduler-sim/fake_rqd.py

```
@@ -0,0 +1,280 @@
+    with _heap_lock:
+        return _alive.pop(frame_id, None)
+
+
+def _send_completion(frame, due_time, killed=False):
+    # A real OOM kill (killed) reports a memory failure; otherwise inject one at
+    # the configured rate. Either way cuebot raises the layer's memory and retries.
+    mem_fail = killed or (_MEM_FAILURE_RATE > 0 and random.random() < _MEM_FAILURE_RATE)
+    exit_status = _EXIT_MEM_FAILURE if mem_fail else 0
+    # A license denial is not a memory failure, so it only applies to frames that
+    # were not already failing for memory.
+    if not mem_fail and _LIC_DENY_RATE > 0 and random.random() < _LIC_DENY_RATE:
```

Ruff (0.16.1)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)